### ISSUE TITLE
Upload .deb and .rpm to packagecloud

### ISSUE THREAD

**Issue by ncw** [rclone/rclone#2153]
Upload the built .rpm and .deb to packagecloud.io

**Comment by lskillen** [comment]
@ncw <3 rclone - Please feel free to consider [Cloudsmith](https://cloudsmith.io) as well for packaging. :-)

It's also commercial but has very generous free plans and offerings for open-source projects.

*Full disclosure*: I work for Cloudsmith, so might be biased (but only _slightly_, honest). We use rclone too!

**Comment by lskillen** [comment]
@ncw Still interested? We added the support for "agnostic" distributions, so that you can upload packages that work for "any distribution", "any version" and "any architecture". You can also divide this into the stable and beta streams.

**Comment by AzureRaptor** [comment]
@ncw I hope you can take a look at this.  Having to install (and "maintain") things via a 'curl'ed shell script is _reeeeallly_ kinda 1999.  ;}

**Comment by azlux** [comment]
My personal repo is already set for that (apt only). Step to test it : 
```
echo "deb http://packages.azlux.fr/debian/ buster main" | sudo tee /etc/apt/sources.list.d/azlux.list
wget -qO - https://azlux.fr/repo.gpg.key | sudo apt-key add -
sudo apt update
sudo apt install rclone
```

A discussion is already in progress on the forum for rlcone to have is own repository and/or put unofficial repo into the docs : https://forum.rclone.org/t/apt-repository-for-rclone/13725/7

@AzureRaptor I don't like manual script to update soft on all my machine. Too touchy.

**Comment by AzureRaptor** [comment]
@azlux Thank you kindly for this information!

I agree completely about manual scripts and maintenance of any kind.  I strongly prefer to use the distro's preferred tools for updating software and system maintenance, than a hodgepodge of sloppy methods I constantly have to tweak, babysit or juryrig.

**Comment by AzureRaptor** [comment]
@azlux  Firstly I want to just thank you again for providing an apt repo for everyone! 

But, something seems to be up with https://packages.azlux.fr/  ... when I go there I see:
(aptitude also yields a similar error)

Secure Connection Failed

An error occurred during a connection to azlux.fr. Peer reports incompatible or unsupported protocol version. Error code: SSL_ERROR_PROTOCOL_VERSION_ALERT

    The page you are trying to view cannot be shown because the authenticity of the received data could not be verified.
    Please contact the website owners to inform them of this problem.


Maybe there's some problem going on with your hosting provider?

**Comment by azlux** [comment]
@AzureRaptor what's your OS version ? This SSL certificate is a "let's encrypt" one. So you need quite recent OS with up-to-date Certificates database.
In case you cannot upgrade, you can use the repo in `http` , everything is signed so you will no have man-in-the-middle<

**Comment by AzureRaptor** [comment]
@azlux Right now I'm on Ubuntu 16.04.  The error also happens on Windows 10, though.
Never occurred to me this might be an OS/browser-specific issue.  That seems kind of odd...I've added certificates for websites before and this wasn't that kind of error.

Is this a recent change to your website?  This error only started showing up tonight.  I've been using your repo for months previously without error.

Seems like something else may be misconfigured anyway.  I tried to go to http://url on your website and it just redirected to https.

**Comment by azlux** [comment]
@AzureRaptor You make a point, I've make a migration yesterday and all my http is redirected to https. It's fixed.

**Comment by AzureRaptor** [comment]
@azlux Is there something else I could install/update in my browser/os, a certificate, or SSL library or something that might be out of date to get https working though?  This still seems a little hinky, like something's misconfigured at one of our ends.  I have no other problem with any other repo or website, and I'm sure others are probably using Let's Encrypt/certbot certs.  Are there any other errors in your migration logs, or cert renewal logs or anything?

**Comment by azlux** [comment]
@AzureRaptor yes, my nginx have update the SSL config from [Modern SSL config](https://wiki.mozilla.org/Security/Server_Side_TLS#Modern_compatibility) . Since few weeks this rule have remove TLS1.2 support. I've rollback to Intermediate compatibility how enable TLS 1.2 and TLS 1.3

**Comment by AzureRaptor** [comment]
@azlux I really appreciate you taking the time to look into this.  I'm honestly surprised the browser & aptitude I was using doesn't seem to support TLS 1.3; I'll look into that.  But if you think Intermediate will be secure enough for you, in the meantime that makes everything work 100% for me; just checked.  Thank you again.

**Comment by ivandeex** [comment]
A PPA on launchpad.net would be a free alternative for .deb

**Comment by ivandeex** [comment]
According to https://packagecloud.io/pricing they _could_ provide free deb/rpm hosting on request. Can't find any info about OSS limits in https://packagecloud.io/docs#faq

**Comment by lskillen** [comment]
As mentioned above, @cloudsmith-io is happy to offer OSS hosting. It's 10GB/100GB for storage/bandwidth by default, but we can go a lot higher for sponsored projects. :-)

**Comment by CoolCold** [comment]
to summarize, what is progress here? any official repo is set? on Ubuntu 18.04 rclone fails to connect to SFTP service, which I'd consider one of the core functionalities!

**Comment by cr1xu5** [comment]
For those who cannot wait, here is another Debian repository: https://gitlab.com/cr1xu5/rclone-debian-repository

It's hosted for free on Gitlab Pages and contains only rclone. Updated automatically each day to the latest stable version.

**Comment by ncw** [comment]
> For those who cannot wait, here is another Debian repository: https://gitlab.com/cr1xu5/rclone-debian-repository
> 
> It's hosted for free on Gitlab Pages and contains only rclone. Updated automatically each day to the latest stable version.

That is very nice @cr1xu5 . I see you using repropro in there - I had an experiment with that - it was rather complicated!

Could this be ported to a GitHub action? It could then be hung off the normal and beta releases.

**Comment by cr1xu5** [comment]
Depends on your goals. When self-hosting the repository you could also offer the older rclone versions at the same time with [the reprepro fork I'm using](https://gitlab.com/packaging/reprepro-multiple-versions).

The Github action then only needs to run the `reprepro includedeb` command for each .deb on the downloads/bèta server. The Debian files are already on that server if I'm correct.
